commands/createfixpullrequests.go

```diff
@@ -307,42 +316,55 @@ func (cfp *CreateFixPullRequestsCmd) updatePackageToFixedVersion(packageType cor
-	case coreutils.Yarn:
-		commandArgs := []string{"up"}
-		err = fixPackageVersionGeneric(packageType.GetExecCommandName(), commandArgs, impactedPackage, fixVersion, "@")
 	case coreutils.Pip:
 		err = fixPackageVersionPip(impactedPackage, fixVersion, requirementsFile)
-	case coreutils.Pipenv:
-		commandArgs := []string{"install"}
-		err = fixPackageVersionGeneric(packageType.GetExecCommandName(), commandArgs, impactedPackage, fixVersion, "==")
 	case coreutils.Poetry:
 		err = fixPackageVersionPoetry(impactedPackage, fixVersion)
+	case coreutils.Go:
+		err = fixPackageVersionGo(impactedPackage, fixVersion)
 	default:
-		return fmt.Errorf("package type: %s is currently not supported", string(packageType))
+		err = fixPackageVersionGeneric(packageType, impactedPackage, fixVersion)
 	}
-
 	return
 }
 
 // The majority of package managers already support upgrading specific package versions and update the dependency files automatically.
 // In other cases, we had to handle the upgrade process
-// commandName - Name of the package manager
-// commandArgs - Package manager upgrade command
+// technology - Name of the package manager
 // impactedPackage - Vulnerable package to upgrade
 // fixVersion - The version that fixes the vulnerable package
-// operator - The operator between the impactedPackage to the fixVersion
-func fixPackageVersionGeneric(commandName string, commandArgs []string, impactedPackage, fixVersion, operator string) error {
+func fixPackageVersionGeneric(technology coreutils.Technology, impactedPackage, fixVersion string) error {
+	commandArgs := []string{technology.GetPackageInstallOperator()}
+	operator := technology.GetPackageOperator()
 	fixedPackage := impactedPackage + operator + fixVersion
+
 	commandArgs = append(commandArgs, fixedPackage)
-	return runPackageMangerCommand(commandName, commandArgs)
+	return runPackageMangerCommand(technology.GetExecCommandName(), commandArgs)
+}
+
+// Module paths in GO must have a major version suffix like /v2 that matches the major version.
+// For example, if a module has the path example.com/mod at v1.0.0, it must have the path example.com/mod/v2 at version v2.0.0.
+// Also bear in mind, that GitHub uses "/" while gopkg use "." to indicate major version change
+// Further reading https://github.com/golang/go/wiki/Modules#semantic-import-versioning
+func handleGoPackageSemanticVersionSuffix(impactedPackage, fixVersion string) (string, error) {
+	pathSeparator := "/"
+	shouldMentionFirstVersion := false // Weather or not to mention /v1 packages
+	majorVersion, err := strconv.Atoi(strings.Split(fixVersion, semanticVersioningSeparator)[0])
+	importPathPrefix := strings.Split(impactedPackage, ".")[0]
+	if err != nil {
+		return "", err
+	}
+	if importPathPrefix == gopkg {
+		pathSeparator = "."
+		shouldMentionFirstVersion = true
+	}
+	if majorVersion > 1 || shouldMentionFirstVersion {
+		impactedPackageWithoutVersion := strings.Split(impactedPackage, pathSeparator+"v")[0]
```

Comment:
`http://github.com/vbauerster/mpb/v7` will not work in that case
It will split 'http://github.com', 'bauerster/mpb' , 'v7'

Reply:
Great edge case ! i've fixed it and added a test to verify this cases.